### PR TITLE
Add workflow to deploy plugin to WordPress.org

### PR DESCRIPTION
**Pull request by attackant**
- Fixes [1016](https://github.com/alleyinteractive/apple-news/issues/1016)

### CODE REVIEW

File: .github/workflows/deploy-to-wordpress.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+name: Deploy to WordPress.org
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - develop
```
Comment (renatonascalves) [inline .github/workflows/deploy-to-wordpress.yml]: 🤔 Do you want to deploy whenever a commit is merged to the main branch?

Looking at the rest of the file, you want to do that when creating a new tag/release.

File: .github/workflows/deploy-to-wordpress.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build
+        run: npm run build
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.0
```
Comment (renatonascalves) [inline .github/workflows/deploy-to-wordpress.yml]:
```suggestion
          php-version: 8.2
```

File: .github/workflows/deploy-to-wordpress.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+      - name: Install PHP dependencies
+        uses: ramsey/composer-install@v2
+        with:
+          composer-options: "--no-dev --optimize-autoloader"
+
+      - name: Get the version
+        id: packagejson
+        run: |
+          echo "::set-output name=version::$(jq -r '.version' package.json)"
```
Comment (renatonascalves) [inline .github/workflows/deploy-to-wordpress.yml]: 🤔 Could you elaborate on why we are getting the version from the `package.json`? I'd assume this is not changed every time, no?

File: .github/workflows/deploy-to-wordpress.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ steps.packagejson.outputs.version }}
+          release_name: Release v${{ steps.packagejson.outputs.version }}
+
+      - name: Create artifact
+        run: |
+          mkdir plugin-build
+          composer archive -vvv --format=zip --file="plugin-build/publish-to-apple-news"
```
Comment (renatonascalves) [inline .github/workflows/deploy-to-wordpress.yml]: I'd suggest reading this https://github.com/10up/action-wordpress-plugin-deploy?tab=readme-ov-file#excluding-files-from-deployment

This project doesn't have the files supported. That means many files would be deployed to WordPress, things you might now want to bundle in the deployed package.

Here is an example of how you can remove them: https://github.com/renatonascalves/wp-graphql-buddypress/blob/main/composer.json#L114-L132

Or you can add one of the deploy files supported by the action.

**Review by renatonascalves — DISMISSED** [review]

File: .github/workflows/deploy-to-wordpress.yml (hunk 1 above)
Comment (renatonascalves) [reply .github/workflows/deploy-to-wordpress.yml]: Alley has Github actions to create automatic tag releases. You might want to use them and leave this only to deploy the plugin to WordPress.

File: .github/workflows/built-release.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+      - main
+      - production
+
+jobs:
+  built-release:
+    uses: alleyinteractive/.github/.github/workflows/built-release.yml@main
+    if: ${{ github.repository != 'alleyinteractive/create-wordpress-plugin' }}
+    with:
+      node: 16
+      php: '8.2'
```
Comment (srtfisher) [inline .github/workflows/built-release.yml:16]: This seems like it was copied over from create-wordpress-plugin. Just want to make sure these are the proper versions.

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "publish-to-apple-news",
-  "version": "2.4.0",
+  "version": "2.4.1",
```
Comment (srtfisher) [inline package.json:3]: Bumping the plugin version here will make the plugin version out of step with the plugin's headers (which WordPress uses to manage the version). To mirror `create-wordpress-plugin`, we should also update the plugin's headers here:

- https://github.com/alleyinteractive/apple-news/blob/develop/apple-news.php#L17
- https://github.com/alleyinteractive/apple-news/blob/develop/readme.txt#L8

That is what the `npm run release` command should do, too.

**Review by srtfisher — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (attackant) [reply package.json:3]: on it

File: .github/workflows/built-release.yml (hunk 5 above)
Comment (attackant) [reply .github/workflows/built-release.yml:16]: Yes, those are correct, we have ticketed updating node for later.

**Review by srtfisher — APPROVED** [review]

**Pull request merged**